Ruddarr/Utilities/WithDefault.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+
+protocol SimpleInitializable {
+    init()
+}
+protocol DefaultValueProvider {
+    associatedtype Value
+    static var defaultValue: Value { get }
+}
+struct SimpleDefaultValueProvider<Value>: DefaultValueProvider where Value: SimpleInitializable {
+    static var defaultValue: Value { .init() }
+}
+
+@propertyWrapper
+struct WithDefault<DefaultProvider: DefaultValueProvider> {
+    typealias Value = DefaultProvider.Value
+    var wrappedValue: Value
+}
+typealias WithSimpleDefault<Value> = WithDefault<SimpleDefaultValueProvider<Value>> where Value: SimpleInitializable
```

Comment:
this is complex to prove out that the approach can support arbitrary default values (by defining arbitrary providers). If we only need empty array, it can be simplified. Let me know.